### PR TITLE
Issue #3162491 by ronaldtebrake: Update Group from rc-5 to 1.4

### PR DESCRIPTION
**Pull request by ronaldtebrake**
## Problem
Ensure Group is up to date and D9 compatible.

## Solution
1. Upgrade from rc-5 to group 1.4 
Including:

https://www.drupal.org/project/group/releases/8.x-1.0
https://www.drupal.org/project/group/releases/8.x-1.1
https://www.drupal.org/project/group/releases/8.x-1.2
https://www.drupal.org/project/group/releases/8.x-1.3
https://www.drupal.org/project/group/releases/8.x-1.4

2. Move patches for gvbo / ginivite / grequest to Open Social and make them D9 compatible.

3. Update baseline for the above so we can start improving where necessary.

4. Create follow up to move Drupal/core to 9.x. 
Otherwise we will run this PR with Core 9.2 as this is the last dependency to update. This results in issues with phpunit which are not related to this PR so rather do that separately.

5. Create follow up to deprecate gvbo/ginvite/grequest out.

6. Ensure we use the gnode access system still to minimize regression, we will move to entity access for 12.0.0

## Issue tracker
https://www.drupal.org/project/social/issues/3162491#comment-14268656

## How to test
Test all Group member management features for Site manager / Group manager
- [ ] As a SM & GM I need to be able to send emails to all members
- [ ] As a SM & GM I need to be able to add / invite new members
- [ ] As a SM & GM I need to be able to accept/deny membership requests
- [ ] As a SM & GM I need to be able to export members of my group

Test all Group content related features:
- [ ] As a SM & Event Organizer I need to be able to manage my event also if it's placed in a Group
- [ ] Make sure to also add a new organizer instead of testing it only as node author
- [ ] Make sure as LU you can comment on content in a Group you are a member of
- [ ] Make sure as LU you receive notifications from your Group content

## Release notes
Group has been updated to 8.x-1.4.

## Change Record
Please take a careful look at the changenotes related to [8.x-1.0](https://www.drupal.org/list-changes/group/published?keywords_description=&to_branch=&version=8.x-1.0&created_op=%3E%3D&created%5Bvalue%5D=&created%5Bmin%5D=&created%5Bmax%5D=)

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2574-6lkkp6owiwzokravv7bogjmzsbmlejw8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61810383c0bc915df7cdf152

File: modules/social_features/social_group/src/SetGroupsForNodeService.php [hunk 1]
```diff
@@ -178,7 +181,7 @@ public static function addGroupContent(NodeInterface $node, Group $group) {
    */
   public static function removeGroupContent(NodeInterface $node, Group $group) {
     // Try to load group content from entity.
-    if ($group_contents = GroupContent::loadByEntity($node)) {
+    if (($group_contents = GroupContent::loadByEntity($node)) && !empty($group_contents)) {
```
Comment (tbsiqueira) [inline modules/social_features/social_group/src/SetGroupsForNodeService.php:184]: Ronald, I believe that we could take the same approach here, and move the assign of $group_contents to outside the if, just to be 100% safe.

**Review by tbsiqueira — APPROVED** [review]
Thank you all for the great effort in making this PR happen. 

Let's merge it to main and work the small improvement from modules/social_features/social_group/src/SetGroupsForNodeService.php at the next one.

**Pull request merged**